lib/shared/serialization.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @fileoverview Object serialization utils.
+ * @author Bryan Mishkin
+ */
+
+"use strict";
+
+const isPlainObject = require("lodash.isplainobject");
+
+/**
+ * Check if a value is a primitive or plain object created by the Object constructor.
+ * @param {any} val the value to check
+ * @returns {boolean} true if so
+ * @private
+ */
+function isSerializablePrimitiveOrPlainObject(val) {
+    return (val === null || typeof val === "string" || typeof val === "boolean" || typeof val === "number" || Array.isArray(val) || isPlainObject(val));
+}
```

Comment:
should check array/object recusively:
```js
function isSerializablePrimitiveOrPlainObject(val) {
  if (
    val === null ||
    val === undefined ||
    typeof val === "string" ||
    typeof val === "boolean" ||
    typeof val === "number"
  ) {
    return true;
  }
  if (Array.isArray(val)) {
    return val.every((item) => isSerializablePrimitiveOrPlainObject(item));
  }
  if (val.constructor === Object) {
    return Object.values(val).every((item) =>
      isSerializablePrimitiveOrPlainObject(item),
    );
  }
  return false;
}

```

Reply:
I'm not sure if that's necessary since `isSerializable` does its own recursion. I have test cases for nested arrays/objects in `tests/lib/shared/serialization.js`. Do you see any test cases I haven't covered yet?

Reply:
well, I missed it was checked in the `isSerializable()`. so the above implementation is same as your `isSerializable()`

just 2 minor suggestions:

1.  we can just use `val.constructor === Object`, to avoid introduce `lodash.isplainobject`.
2. should explictly check `undefined` - it's not serializable.

Reply:
1. I removed lodash.isplainobject. Note that it does check some [extra stuff](https://github.com/lodash/lodash/blob/main/src/isPlainObject.ts) but perhaps `val.constructor === Object` for us.
2. I have a test case that `undefined` is not serializable, whereas `null` is.